spec/shared/base_strategy.rb

```diff
@@ -0,0 +1,36 @@
+require "spec_helper"
+
+shared_examples_for Brainstem::QueryStrategies::BaseStrategy do
+  let(:strategy) { described_class.new(options) }
+
+  describe "#calculate_per_page" do
+    let(:result) { strategy.calculate_per_page }
+
+    [
+      # per_page  default_per_page  max_per_page  default_max_per_page  expected   message
```

Comment:
This is pretty cool

Reply:
A nice to have, and maybe something for next time would be dynamic `it` descriptions. Here we have dynamic describes, ie. `no per page and default > max` but then it would be nice to add something for the `it`, ie. `it uses max per page`.

Reply:
That's a good call, @bennapp. I'll look at adding that, as we have a day or two before deploying this